### PR TITLE
[core]Add schema validation for record-level.time-field

### LINKED ISSUE #4724: [Bug] Time field for record-level expire may block compaction
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Paimon version

0.9.0

### Compute Engine

Flink

### Minimal reproduce step

```sql
CREATE TABLE T (pk INT, a INT, ts INT) 
WITH (
  'bucket'='1', 
  'primary-key'='pk', 
  'full-compaction.delta-commits' = '1', 
  'record-level.expire-time'='1s', 
  'record-level.time-field'='ts')
```

```sql
-- batchsql 1, no compact, "dirty data"
INSERT INTO T VALUES (1, 1, CAST(NULL AS INT));

-- batchsql 2，trigger compact
INSERT INTO T VALUES (2, 2, 2);    -- ERROR: Time field for record-level expire should not be null.
```

### What doesn't meet your expectations?

When a row of dirty data (the value of time field for record-level expire is null) writed into L0 files, the following compaction will fail. I think it is a dangerous phenomenon, if there is no snapshot to roll back to. 
So should there be stricter constraints on this time field? For example, when creating a table, make sure that the type of this field is not nullable.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by herefree**
### Purpose
Add schema validation for record-level.time-field

Linked issue: close #4724

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java [hunk 1]
```diff
@@ -176,6 +181,35 @@ public static void validateTableSchema(TableSchema schema) {
             }
         }
 
+        if (schema.options().containsKey(CoreOptions.RECORD_LEVEL_TIME_FIELD.key())) {
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java:184]: String recordLevelTimeField = options.recordLevelTimeField();
if (recordLevelTimeField != null) {
    xxx;
}

File: paimon-core/src/test/java/org/apache/paimon/schema/SchemaValidationTest.java [hunk 2]
```diff
@@ -103,4 +104,22 @@ public void testFromSnapshotConflict() {
                 .hasMessageContaining(
                         "[scan.snapshot-id] must be null when you set [scan.timestamp-millis,scan.timestamp]");
     }
+
+    @Test
+    public void testRecordLevelTimeField() {
+        Map<String, String> options = new HashMap<>();
```
Comment (wwj6591812) [inline paimon-core/src/test/java/org/apache/paimon/schema/SchemaValidationTest.java:110]: Map<String, String> options = new HashMap<>(2);

File: paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java [hunk 3]
```diff
@@ -19,17 +19,49 @@
...
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import java.time.Duration;
+import java.util.UUID;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 class RecordLevelExpireTest extends PrimaryKeyTableTestBase {
+    @Override
```
Comment (wwj6591812) [inline paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java:43]: Please add a blank line before @Override.

File: paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java [hunk 4]
```diff
@@ -19,17 +19,49 @@
...
+    @BeforeEach
+    public void beforeEachBase() throws Exception {
```
Comment (wwj6591812) [inline paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java:45]: Why add this method?

Comment (herefree) [reply paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java:45]: Column "col1" datatype is `DataTypes.INT()` in PrimaryKeyTableTestBase, but it should be `DataTypes.INT().notNull()` in RecordLevelExpireTest.

File: paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java (hunk 1 above)
Comment (herefree) [reply paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java]: fixed

File: paimon-core/src/test/java/org/apache/paimon/schema/SchemaValidationTest.java (hunk 2 above)
Comment (herefree) [reply paimon-core/src/test/java/org/apache/paimon/schema/SchemaValidationTest.java]: fixed

File: paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java (hunk 3 above)
Comment (herefree) [reply paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java:44]: fixed

File: paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java (hunk 4 above)
Comment (wwj6591812) [reply paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireTest.java:46]: ok

**Comment by wwj6591812** [comment]
+1

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**